mesonbuild/backend/ninjabackend.py

```diff
@@ -1255,6 +1250,26 @@ def generate_rust_target(self, target, outfile):
             self.generate_shsym(outfile, target)
         self.create_target_source_introspection(target, rustc, args, [main_rust_file], [])
 
+    @staticmethod
+    def get_rule_suffix(for_machine: MachineChoice) -> str:
+        return PerMachine('_FOR_BUILD', '', '_FOR_TARGET')[for_machine]
```

Comment:
I'm not sure it's worth the runtime cost of instantiating a PerMachine and doing a lookup to avoid:
```python
if for_machine is MachineChoice.BUILD:
    return '_FOR_BUILD'
elif for_machine is MachineChoice.HOST:
     return ''
return '_FOR_TARGET'
```

It's really too bad python doesn't have a case statement.

Reply:
I actually think `MachineChoice` code change. Having multiple hosts and target machines makes perfect sense, for example. So I rather do this for the completion checking.

Reply:
To do that you're going to have to replace PerMachine anyway since it already assumes that you have a single build, a single host, and a single target. If you decide to do that kind of refactor this is going to be the least of your worries.

Reply:
Right I mean because I have to change `PerMachine` this will blow up if I forget to change it too.

Reply:
Can you run a test for me then? Write the if statement I proposed and and the function as-is in this MR and use timeit to see how long they take? If the `PerMachine` instantiation and lookup aren't too much more expensive than the if would be then I'll let this go.

Reply:
Sure. Any idea what test `meson.build` would be good to use? Sadly https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/mesa/default.nix doesn't yet use the meson build system, but I can fix that :).